combined_robot_control:
  robot_hardware:
    - dynamixel_control
  loop_frequency: 50
  cycle_time_error_threshold: 0.1
  dynamixel_control:
    type: dynamixel/hardware_interface_p1
    serial_interface: /dev/ttyACM0
    baudrate: 1000000 # in bauds
    read_timeout: 0.05 # in seconds
    scan_timeout: 0.05 # in seconds
    # Cofniguration of the servos
    servos:
      arm_joint_1:
        # hardware ID of the actuator
        id: 2
        # offset to be added, in radians, to the position of an actuator
        offset: 4.5 # in rad
        command_interface: velocity
      arm_joint_2:
        id: 1
        # max joint speed
        # it shoud NOT be used for actuators in velocity mode
        max-speed: 4 # in rad/s
        offset: 3.1415
    # default mode
    default_command_interface: position

dynamixel_controllers:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 10

  pos_controller:
    type: position_controllers/JointPositionController
    joint: arm_joint_2

  vel_controller:
    type: velocity_controllers/JointVelocityController
    joint: arm_joint_1
  

